File: Java_project_workspace/com.maharshi_baba/src/main/java/com/ecommerce/respository/CategoryDao.java
package com.ecommerce.respository;

import com.ecommerce.vo.CategoryVo;

import java.util.ArrayList;

public interface CategoryDao {

    public void insertCategory(CategoryVo categoryVo);
    public ArrayList<CategoryVo> searchCategory();
    public ArrayList<CategoryVo> findByIdCategory(int id);


}

File: Python_Tution_Flask_workspace/flask_project_fp_/base/com/dao/category_dao.py
from base import db
from base.com.vo.category_vo import CategoryVo


class CategoryDao:
    def insert_category(self, category_vo):
        db.session.add(category_vo)
        db.session.commit()

    def search_category(self):
        query = db.session.query(CategoryVo).all()
        db.session.commit()
        return query

    def edit_category(self, edit_id):
        query = db.session.query(CategoryVo).filter_by(category_id=edit_id)
        db.session.commit()
        return query

    def update_category(self, category_vo):
        db.session.merge(category_vo)
        db.session.commit()

    def delete_category(self, delete_category_vo_obj):
        db.session.delete(delete_category_vo_obj)
        db.session.commit()
